File: android/app/src/main/java/com/rafaeltmbr/stopwatch/infra/presentation/view_models/LapsViewModel.kt
package com.rafaeltmbr.stopwatch.infra.presentation.view_models

import com.rafaeltmbr.stopwatch.domain.entities.Status
import com.rafaeltmbr.stopwatch.infra.presentation.entities.ViewLap
import kotlinx.coroutines.flow.StateFlow


interface LapsViewModel {
    data class State(
        val status: Status = Status.INITIAL,
        val time: String = "00:00.00",
        val laps: List<ViewLap> = emptyList(),
    )

    sealed class Action {
        data object Resume : Action()
        data object Lap : Action()
        data object NavigateBack : Action()
    }

    val state: StateFlow<State>
    fun handleAction(action: Action)
}

File: ios/Stopwatch/infra/presentation/view_models/LapsViewModel.swift
import Foundation

struct LapsState {
    let status: Status
    let milliseconds: String
    let laps: [ViewLap]
    
    init(status: Status = .initial, milliseconds: String = "00:00.00", laps: [ViewLap] = []) {
        self.status = status
        self.milliseconds = milliseconds
        self.laps = laps
    }
}

enum LapsAction {
    case resume, lap
}

protocol LapsViewModel: ObservableObject {
    var state: LapsState { get }
    
    func handleAction(_ action: LapsAction)
}
